src/deluge/gui/context_menu/audio_input_selector.h

```diff
@@ -21,15 +21,28 @@
 
 class AudioOutput;
 
+namespace deluge::gui::context_menu {
+
 class AudioInputSelector final : public ContextMenu {
+	enum class Value;
+
 public:
-	AudioInputSelector();
-	bool getGreyoutRowsAndCols(uint32_t* cols, uint32_t* rows);
-	void selectEncoderAction(int8_t offset);
+	AudioInputSelector() = default;
+	bool getGreyoutRowsAndCols(uint32_t* cols, uint32_t* rows) override;
+	void selectEncoderAction(int8_t offset) override;
 	bool setupAndCheckAvailability();
-	bool canSeeViewUnderneath() { return true; }
+	bool canSeeViewUnderneath() override { return true; }
 
 	AudioOutput* audioOutput;
+
+	// Title
+	char const* getTitle() override;
+
+	// Options
+	size_t getNumOptions() override;
+	const char** getOptions() override;
```

Comment:
Can we C++-ify this somehow? Having 2 separate functions sounds like a great trap for young players.

Reply:
The best way to do that would probably be returning a std::pair of the `char**`, and a `size_t` that can then be destructured in the caller...